LabExT/Movement/Stage3DSmarAct.py

```diff
@@ -0,0 +1,625 @@
+
+        @property
+        def speed(self) -> float:
+            """Returns speed setting of channel in micrometers/seconds specified by SA_GetClosedLoopMoveSpeed_S"""
+            system_speed = ct.c_int()
+            if self._stage._exit_if_error(
+                MCSC.SA_GetClosedLoopMoveSpeed_S(
+                    self._stage.handle,
+                    self._handle,
+                    system_speed)):
+                self._speed = self._to_mircometer(system_speed.value)
+
+            return self._speed
+
+        @speed.setter
+        def speed(self, umps: float) -> None:
+            """Sets speed of channel in micrometers/seconds by calling SA_SetClosedLoopMoveSpeed_S
+
+            Parameters
+            ----------
+            umps : float
+                Speed measured in um/s
+            """
+            if self._stage._exit_if_error(MCSC.SA_SetClosedLoopMoveSpeed_S(
+                self._stage.handle,
+                self._handle,
+                ct.c_int(int(self._to_nanometer(umps)))
+            )):
+                self._speed = umps
+
+        @property
+        def acceleration(self) -> float:
+            """Returns acceleration of channel in micrometers/seconds^2 specified by SA_GetClosedLoopMoveAcceleration_S"""
+            system_acceleration = ct.c_int()
+            if self._stage._exit_if_error(
+                MCSC.SA_GetClosedLoopMoveAcceleration_S(
+                    self._stage.handle,
+                    self._handle,
+                    system_acceleration)):
+                self._acceleration = system_acceleration.value
+
+            return self._acceleration
+
+        @acceleration.setter
+        def acceleration(self, umps2: float) -> None:
+            """Sets acceleration of channel in micrometers/seconds^2 by calling SA_SetClosedLoopMoveAcceleration_S
+
+            Parameters
+            ----------
+            umps2 : float
+                Acceleration measured in um/s^2
+            """
+            if (self._stage._exit_if_error(
+                    MCSC.SA_SetClosedLoopMoveAcceleration_S(
+                        self._stage.handle,
+                        self._handle,
+                        ct.c_int(int(umps2))
+                    ))):
+                self._acceleration = umps2
+
+        @property
+        def movement_mode(self) -> MovementType:
+            """Returns movement mode of channel: Either RELATIVE or ABSOLUTE"""
+            return self._movement_mode
+
+        @movement_mode.setter
+        def movement_mode(self, mode: MovementType) -> None:
+            """Set movement mode of channel: Either RELATIVE or ABSOLUTE"""
+            if mode.value not in MovementType._value2member_map_:
+                raise ValueError("Invalid movement mode {}".format(str(mode)))
+            self._movement_mode = mode
+
+        # Movement
+
+        def move(
+                self,
+                diff: float,
+                mode=MovementType.RELATIVE,
```

Comment:
I would make this a mandatory argument, so to force the programmer to specify it.

Reply:
Good point, I agree. Will change it 👍🏼